Fixing issue with leaving dirty input forms

muros-ct opened the pull request:
There are forms on UI that can be left in dirty state by using main
menu - West menu.
This was solved by adding dirty form flag on gwtSession object and
when user uses West menu it is notified that changes will be lost and
can decide if this is ok or not.

Changes were required to WestNavigationView and that could break unexpected
functionality of UI. It was tested but not overall UI tested.

This fixes issue #1137.

Signed-off-by: Uros Mesaric Kunst <uros.mesaric-kunst@comtrade.com>

muros-ct commented:
@lorthirk Please take a look as I am changing gwtSession and WestNavigationView.
@LeoNerdoG will check functionality and see if I broke something.

LeoNerdoG commented:
@muros-ct 
I have tested this fix and found a problem with some fields (I am using Raspberry Pi 3 with Kura 3.1.1).
I still haven't finished testing, but if I change for example **Clock.ntp.host** field, Kapua does not open the confirmaiton field, but if I change **Clock.ntp.port** field, the confirmation window pops-up. 
Can you please check this?

muros-ct commented:
It probably has to do with field loosing focus and because field doesn't have validator, field is not marked as changed. If field is changed and moved to other field it still works, or if field such as radio is changed.
I will look intoi it and try to solve it if at all possible.

muros-ct commented:
I pushed changes that solve issue with some field changes not invoking confirmation dialog.
I fixed on TextField and TextArea (available on Camel rules configuration).
@LeoNerdoG please evaluate.
@lorthirk Please see changes on KapuaTextField and added KapuaTextArea.

muros-ct commented:
I looked at TriggerField implementation that is using triggerBlur.
mimicing prevents onFocus to trigger muliple focusEventPreview.

LeoNerdoG commented:
I have verified again and now also the password fields are OK. You can merge when ready.

muros-ct commented:
@lorthirk Is it ok if I merge?

lorthirk commented:
Ok for me. @Coduz do you agree?
